packages/@aws-cdk/aws-servicecatalogappregistry/lib/target-application.ts

```diff
@@ -117,7 +119,11 @@ class ExistingTargetApplication extends TargetApplication {
     super();
   }
   public bind(scope: Construct): BindTargetApplicationResult {
+    const arnComponents = cdk.Arn.split(this.applicationOptions.applicationArnValue, cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME);
+    const applicationId = arnComponents.resourceName;
     const stackId = this.applicationOptions.stackId ?? 'ApplicationAssociatorStack';
```

Comment:
Since it looks like `stackId` originally existed to change the stack name (but was the wrong mechanism for that), can we also deprecate `stackId`?

Add: 

```
  * @deprecated Use `stackName` instead to control the name of the stack
```

To its docstring?

Reply:
Yes this will be more helpful to the user. I'll add a `deprecated` statement to the docstring in the next commit.